zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationPreconditions.java

```diff
@@ -451,20 +452,37 @@ public static void requireSameElementType(
       final String targetElementId,
       final ProcessInstanceRecord elementInstanceRecord,
       final long processInstanceKey) {
-    final BpmnElementType targetElementType =
+    BpmnElementType targetElementType =
         targetProcessDefinition.getProcess().getElementById(targetElementId).getElementType();
-    if (elementInstanceRecord.getBpmnElementType() != targetElementType) {
-      final String reason =
-          String.format(
-              ERROR_ELEMENT_TYPE_CHANGED,
-              processInstanceKey,
-              elementInstanceRecord.getElementId(),
-              elementInstanceRecord.getBpmnElementType(),
-              targetElementId,
-              targetElementType);
-      throw new ProcessInstanceMigrationPreconditionFailedException(
-          reason, RejectionType.INVALID_STATE);
+
+    if (elementInstanceRecord.getBpmnElementType() == targetElementType) {
+      return;
     }
+
+    // if target element is a multi instance body, we should check the inner activity element type
+    // because the inner activity of the multi instance body can still match the source element's
+    // type
+    if (targetElementType == BpmnElementType.MULTI_INSTANCE_BODY) {
+      final ExecutableMultiInstanceBody targetElement =
+          targetProcessDefinition
+              .getProcess()
+              .getElementById(targetElementId, ExecutableMultiInstanceBody.class);
+      targetElementType = targetElement.getInnerActivity().getElementType();
+      if (elementInstanceRecord.getBpmnElementType() == targetElementType) {
+        return;
+      }
+    }
```

Comment:
❌ Please add a test case that shows this case. E.g. a multi-instance marked service task changed to a multi-instance marked user task.